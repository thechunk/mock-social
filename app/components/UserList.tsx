import React, {PureComponent} from 'react';
import UserListCell from "./UserListCell";
import {IWithUserListDataProps} from "../hoc/withUserListData";

class UserList extends PureComponent<IWithUserListDataProps> {
    render() {
        return this.props.users != null
            ? this.props.users.map(v => <UserListCell key={v.id} title={v.name} />)
            : null;
    }
}

export default UserList;