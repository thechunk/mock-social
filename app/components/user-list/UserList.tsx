import React, {PureComponent} from 'react';
import UserListCell from "./UserListCell";
import withUserListData, {IWithUserListDataProps} from "../../hoc/withUserListData";
import {FlatList, GestureResponderEvent} from "react-native";
import {NavigationScreenProps, withNavigation} from "react-navigation";
import {IWithUserDataOptionalProps} from "../../hoc/withUserData";

class UserList extends PureComponent<IWithUserListDataProps & NavigationScreenProps> {
    navigateToDetail(user: IUser) {
        return (event: GestureResponderEvent) => {
            this.props.navigation.navigate('UserDetail',
                { hydrate: user, id: user.id } as IWithUserDataOptionalProps)
        }
    }

    render() {
        return (
            <FlatList data={this.props.users} keyExtractor={v => v.id.toString()} renderItem={v => (
                <UserListCell
                    key={v.index}
                    title={v.item.name}
                    onPress={this.navigateToDetail(v.item)} />
            )} />
        )
    }
}

export default withNavigation(withUserListData(UserList));