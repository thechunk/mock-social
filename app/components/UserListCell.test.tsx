import * as React from 'react';
import jest from 'jest';
import UserList from './UserList';

import {shallow} from "enzyme";

describe('UserList', () => {
    it('renders correctly', () => {
        expect(shallow(<UserList api={jest.mock('../classes/Api')} />)).toMatchSnapshot();
    });
});
