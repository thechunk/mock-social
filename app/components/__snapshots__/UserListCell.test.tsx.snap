// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleSubtitleListCell renders correctly 1`] = `
<ListCell>
  <Text>
    title
  </Text>
</ListCell>
`;
